crypto/x509/t_req.c

```diff
@@ -147,9 +149,15 @@ int X509_REQ_print_ex(BIO *bp, X509_REQ *x, unsigned long nmflags,
                 case V_ASN1_NUMERICSTRING:
                 case V_ASN1_UTF8STRING:
                 case V_ASN1_IA5STRING:
-                    if (BIO_write(bp, (char *)bs->data, bs->length)
-                            != bs->length)
-                        goto err;
+                    if (NID_pkcs9_challengePassword == (OBJ_obj2nid(X509_ATTRIBUTE_get0_object(a)))) {
```

Comment:
RFC2985 defines the challengePassword as follows:

"The challengePassword attribute type specifies a password by which an
   entity may request certificate revocation."

RFC2985 does not define any encryption or other protection for the challengePassword while it is in the CSR. The CA when it receives the CSR will need to access and inspect the challengePassword in order to be able to store it in whatever internal system that CA uses. We can't know how OpenSSL is used by applications, and it may be that this function is used by just such an application involved in just such a process. If so by making this proposed change we would break that application. Therefore I'm not convinced that making this change is a good idea.